src/loader/cmt.ml

```diff
@@ -78,6 +78,10 @@ let rec read_pattern env parent doc pat =
         read_pattern env parent doc pat
     | Tpat_lazy pat ->
         read_pattern env parent doc pat
+#if OCAML_MAJOR = 4 && OCAML_MINOR >= 08
+    | Tpat_exception _ ->
+      []
```

Comment:
That doesn't look right. I think you should recurse in that case.

Reply:
Yes, I think you're right.